UPGRADE.md

```diff
@@ -1,5 +1,13 @@
 # Upgrade to 3.0
 
+## BC BREAK: `OCI8Statement::convertPositionalToNamedPlaceholders()` is removed.
+
+The `OCI8Statement::convertPositionalToNamedPlaceholders()` method has been extracted to an internal utility class.
+
+## BC BREAK: Dropped handling of one-based numeric arrays of parameters in `Statement::execute()`
+
+The statement implementations no longer detect whether `$params` is a zero- or one-based array. A zero-based numeric array is expected.
```

Comment:
We should probably start adding `@psalm-param list<T>` for these

Reply:
Should we add Psalm to the build pipeline first?

Reply:
Sure thing - it's a big amount of work to get it to "like" our codebase though 😅